src/DiffWindow.ts

```diff
@@ -0,0 +1,87 @@
+import JoplinData from "api/JoplinData";
+import JoplinViewsDialogs from "api/JoplinViewsDialogs";
+
+export class DiffWindow {
+    
+    private dialogsApi : JoplinViewsDialogs;
+    private dataApi : JoplinData;
+    private installDir : string;
+    private fileSys;
+    
+    constructor(dilg : JoplinViewsDialogs, data : JoplinData, fileSys, installDir : string) {
+        this.dialogsApi = dilg;
+        this.dataApi = data;
+        this.installDir = installDir;
+        this.fileSys = fileSys;
+    }
+
+
+    private handle: string = "";
+    
+    public async Init(id : string) {
+        this.handle = await this.dialogsApi.create(id);
+		await this.dialogsApi.addScript(this.handle, './lib/codemirror/lib/codemirror.js');
+		await this.dialogsApi.addScript(this.handle, './lib/codemirror/lib/codemirror.css');
+		await this.dialogsApi.addScript(this.handle, './lib/diff_match_patch/diff_match_patch.js');
+		await this.dialogsApi.addScript(this.handle, './lib/codemirror/addon/merge/merge.css');
+		await this.dialogsApi.addScript(this.handle, './ui/DiffWindow/index.js');
+		await this.dialogsApi.addScript(this.handle, './ui/DiffWindow/index.css');
+		await this.dialogsApi.setButtons(this.handle, [
+			{
+				id: 'submit',
+				title: 'Save'
+			},
+			{
+				id: 'cancel',
+				title: 'Cancel'
+			}
+		]);
+    }
+
+    public async OpenWindow(noteId: string, compareWithId : string = "") {
+        const localNote = await this.dataApi.get(['notes', noteId], {
+            fields: ['is_conflict', 'conflict_original_id', 'body', 'title']
+        });
+    
+        if(localNote.is_conflict === 0) {
+            throw new Error('This is not a conflict note.');
+        }
+    
+        if(localNote.conflict_original_id.length === 0 && compareWithId == "") {
+            throw new Error('A note must be specificed to be compared with in case of missing original id.');
+        }
+    
+        const remoteId = (compareWithId == "" ? localNote.conflict_original_id : compareWithId);
+
+        const remoteNote = await this.dataApi.get(['notes', remoteId], {
+            fields: ['body', 'title']
+        });
+    
+        const remoteNoteContent = remoteNote.body.replace(/"/g, '&quot;');
+        const localNoteContent = localNote.body.replace(/"/g, '&quot;');
+        const remoteNoteTitle = remoteNote.title.replace(/"/g, '&quot;');
+        const localNoteTitle = localNote.title.replace(/"/g, '&quot;');
```

Comment:
If you're escaping here, you should make sure to unescape on the other side. It looks like this will mangle notes. You also might want to use an html escaping library rather than performing a simple replacement here.

Reply:
The escaping actually works without unescaping.. Somehow CodeMirror supports html codes and shows them correctly. I'll look into using a proper library for sure though.

Reply:
Have you tested what happens if you save the file and export it? Is CodeMirror un-escaping it, or just rendering it correctly?

Reply:
Hm I can't save/export yet. But I'll keep it in mind.
Viewing though works fine: 
![image](https://user-images.githubusercontent.com/8184424/122435300-2eacfe80-cf98-11eb-8825-00e2b6185798.png)

Reply:
Sounds good, I'm happy with the way it is now, you might find in the future that there are other characters that need to be escaped, that's why a general html escaping library will be best (These libraries exist for exactly this task). 
Since this is already a large PR, you can go ahead and create a github issue to remind yourself to look into a generic html escaper/unescaper.